File: lc_java/lcJava/array/SearchMatrix240.java
package array;


class SearchMatrix240{
    public boolean searchMatrix(int[][] matrix,int target){
        int m = matrix.length,n = matrix[0].length;
        int row = 0,col = n - 1;
        while(row < m && col >= 0){
            if(matrix[row][col] == target){
                return true;
            }else if(target > matrix[row][col]){
                row++;
            }else{
                col--;
            }
        }
        return false;
    }

    public static void main(String[] args) {
        SearchMatrix240 sol = new SearchMatrix240();
        int[][] matrix = {
                {1,4,7,11,15},
                {2,5,8,12,19},
                {3,6,9,16,22},
                {10,13,14,17,24},
                {18,21,23,26,30}
        };
        int target = 20;
        System.out.println(sol.searchMatrix(matrix, target));
    }
}

File: lc_go/lcGo/array/SearchMatrix240.go
package main

import "fmt"

func searchMatrix(matrix [][]int,target int) bool {
	m,n := len(matrix),len(matrix[0])
	row,column := m - 1,0

	for row >= 0 && column < n {
		if matrix[row][column] == target {
			return true
		}else if matrix[row][column] > target {
			row--
		}else {
			column++
		}
	} 
		return false
}

func main(){
	matrix := [][]int{
		{1,4,7,11,15},
                {2,5,8,12,19},
                {3,6,9,16,22},
                {10,13,14,17,24},
                {18,21,23,26,30}
	}
	target := 20
	res := searchMatrix(matrix,target)
	fmt.Println(res)
}